src/Libraries/Microsoft.Extensions.AI.Abstractions/HostedMcpServerToolRequireSpecificApprovalMode.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>
+/// Represents a mode where approval behavior is specified for individual tool names.
+/// </summary>
+public sealed class HostedMcpServerToolRequireSpecificApprovalMode : HostedMcpServerToolApprovalMode
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="HostedMcpServerToolRequireSpecificApprovalMode"/> class that specifies approval behavior for individual tool names.
+    /// </summary>
+    /// <param name="alwaysRequireApprovalToolNames">The list of tools names that always require approval.</param>
+    /// <param name="neverRequireApprovalToolNames">The list of tools names that never require approval.</param>
+    public HostedMcpServerToolRequireSpecificApprovalMode(HashSet<string>? alwaysRequireApprovalToolNames, HashSet<string>? neverRequireApprovalToolNames)
```

Comment:
All of these `HashSet`s should be `ISet`s

Reply:
Do you have any thoughts on https://github.com/dotnet/extensions/pull/6664#discussion_r2283931629?